examples/bri-3/src/shared/prisma/prisma.mapper.ts

```diff
@@ -0,0 +1,161 @@
+import { Injectable } from '@nestjs/common';
+import {
+  BpiAccount as BpiAccountPrismaModel,
+  BpiAccountStateTreeLeafValue as BpiAccountStateTreeLeafValuePrismaModel,
+  BpiMerkleTree as BpiMerkleTreePrismaModel,
+  Message as BpiMessagePrismaModel,
+  BpiSubjectAccount as BpiSubjectAccountPrismaModel,
+  BpiSubject as BpiSubjectPrismaModel,
+  BpiSubjectRole as BpiSubjectRolePrismaModel,
+  Transaction as TransactionPrismaModel,
+  Workflow as WorkflowPrismaModel,
+  Workgroup as WorkgroupPrismaModel,
+  Workstep as WorkstepPrismaModel,
+} from '@prisma/client';
+import { BpiMessage } from '../../bri/communication/models/bpiMessage';
+import { BpiSubjectAccount } from '../../bri/identity/bpiSubjectAccounts/models/bpiSubjectAccount';
+import { BpiSubject } from '../../bri/identity/bpiSubjects/models/bpiSubject';
+import { BpiSubjectRole } from '../../bri/identity/bpiSubjects/models/bpiSubjectRole';
+import { BpiMerkleTree } from '../../bri/merkleTree/models/bpiMerkleTree';
+import { BpiAccount } from '../../bri/state/bpiAccounts/models/bpiAccount';
+import { StateTreeLeafValueContent } from '../../bri/state/models/stateTreeLeafValueContent';
+import { Transaction } from '../../bri/transactions/models/transaction';
+import { Workflow } from '../../bri/workgroup/workflows/models/workflow';
+import { Workgroup } from '../../bri/workgroup/workgroups/models/workgroup';
+import { Workstep } from '../../bri/workgroup/worksteps/models/workstep';
+import { MerkleTreeService } from '../../bri/merkleTree/services/merkleTree.service';
+import MerkleTree from 'merkletreejs';
+
+// We do mapping from prisma models to domain objects using simple Object.assign
+// since automapper is not happy working with types, which is how Prisma generates database entities.
+// For these mappings to work, the convention is that the domain objects have the same properties as the
+// prisma models. In case there is a need to do something custom during mapping, it can be implemented
+// in the appropriate method below.
+
+interface IConstructor<T> {
+  new (...args: any[]): T;
+}
+
+@Injectable()
+export class PrismaMapper {
+  constructor(private readonly merkleTreeService: MerkleTreeService) {}
+
+  public mapBpiSubjectPrismaModelToDomainObject(
```

Comment:
im a bit rusty with typescript, but can these be like generic method, and then only define these verbose methods if there is something more tricky than simple assign? generic like this (this is also ugly, but just as illustration):

```
public map<T extends object>(source: any, targetType: IConstructor<T>): T {
    const target = this.activator(targetType)
    Object.assign(target, source);
    return target;
  }

```

Reply:
Thanks for this, spent some time during POC trying to implement the generic method but got stuck with some dumb issue and decided to leave it for later. Applied now everywhere based on your proposal, please rereview. We can improve as we go. @skosito